src/wing/TiglWingStructureReference.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+* Copyright (C) 2016 Airbus Defense and Space
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "TiglWingStructureReference.h"
+
+#include "CTiglLogging.h"
+//#include "CCPACSTrailingEdgeDevice.h"
+#include "CCPACSWingComponentSegment.h"
+#include "CCPACSWing.h"
+
+
+namespace tigl
+{
+//TiglWingStructureReference::TiglWingStructureReference(CCPACSTrailingEdgeDevice& parent)
+//: type(TrailingEdgeDeviceType), trailingEdgeDevice(&parent) { }
+
+TiglWingStructureReference::TiglWingStructureReference(CCPACSWingComponentSegment& parent)
+: type(ComponentSegmentType), componentSegment(&parent) { }
+
+#define DISPATCH(call) \
+    switch (type) { \
+        case ComponentSegmentType: return componentSegment->call; \
+/*        case TrailingEdgeDeviceType: return trailingEdgeDevice->call; */ \
+        default: std::terminate(); \
```

Comment:
When the default label is reached this means that the type member has an unknown enum value, which means that the implementation is not correct. But you're right, even if the implementation is not correct the application should not stop completely, so i'll replace it by a CTiglError